### ISSUE TITLE
Tax calculation is not following rate in config

### ISSUE THREAD

**Issue by digitweaks** [silvershop/silvershop-core#831]
Ok, so I have SilverShop 4 which works perfectly fine outside of one issue with taxes.

The tax rate defined in the config yml is 21% as it's for Spain and the recorded rate in the table SilverShop_TaxModifier is 0.21 which is correct and even the tax rate displayed is Tax@21%. However, when I calculate the actual tax amount, it's always based on 20% and not 21%. I had that issue already many times which was the reason I didn't use SilverShop for a long time. Now that I'm at the end of a project I realise that issue is still there and even by manually setting the rate directly in the FlatTax class, it stays at 20% whatsoever.

**Comment by wilr** [comment]
@digitweaks Can you share your YAML configuration for review? You should have something like this in your YAML

```
SilverShop\Model\Modifiers\Tax\FlatTax:
  exclusive: false
  rate: 0.21
  name: 'VAT'
```

**Comment by digitweaks** [comment]
Hi @wilr, thanks for the swift reply!

I was a bit far with the YML as I was settings another class SilverShop\Tax\ShopConfigExtension.

However, I will sound stupid... LOL 

I was calculating the VAT at 21% based on the price of the item as if it was exclusive.
So for an item at 30€, I took my calculator, type 30 * 0.21 and it gives me 6.3€... which is false

I tried a rate of 50% just for the sake of the test and obviously the price didn't change... as it's inclusive.

Then I realised that if the price is inclusive, I have to divide the price of the item by 121, multiply by 21
which obviously gives me 5.21€... so there never was an issue, I was just being determined to do the wrong calculation.

I will take the L on this one, I guess being good at mathematics made me too confident in my lack of reflection lol

Thanks a lot for the YML snippet, at least I will stop hardcoding the tax rate in the SilverShop core.
